src/edu/washington/escience/myria/util/DeploymentUtils.java

```diff
@@ -292,7 +293,10 @@ public static void ensureMasterStart(final String hostname, final int restPort,
           break;
         }
         // expected for the first few trials
-        LOGGER.warn("expected exception occurred", e);
```

Comment:
Should save the exception here and, if the master never comes up, print the last one we received. Sometimes the errors are useful!